sandman/tests/test_common_string.cpp

```diff
@@ -0,0 +1,260 @@
+#include "common/string.h"
+
+#include "catch_amalgamated.hpp"
+
+#include <sstream>
+#include <type_traits>
+
+#if false
+inline namespace Debug
+{
+
+	template <typename CharT, std::size_t kN>
+	std::ostream& operator<<(std::ostream& outputStream, Common::String<CharT, kN> const& buffer)
+	{
+		outputStream << "Buffer[";
+		for (typename Common::String<CharT, kN>::Data::size_type i{ 0u }; i < kN; ++i)
+		{
+			if (i == buffer.GetLength()) { outputStream << '@'; continue; }
+			CharT const c{ buffer.GetData().at(i) };
+			switch (c)
+			{
+				case '\0': outputStream << '0'; break;
+				default: outputStream << c; break;
+			}
+		}
+		outputStream << ']';
+		return outputStream;
+	}
+
+	template <typename>
+	static constexpr bool kIsCharBuffer{ false };
+
+	template <typename CharT, std::size_t kN>
+	static constexpr bool kIsCharBuffer<Common::String<CharT, kN>>{ true };
+
+	template <typename T>
+	[[nodiscard]] std::string ToString(T&& object)
+	{
+		std::ostringstream outputStream;
+		outputStream << std::forward<T>(object);
+		return outputStream.str();
+	}
+
+	template <typename... ParamsT>
+	static void Println(ParamsT&&... args)
+	{
+		(Catch::cout() << ... << std::forward<ParamsT>(args)) << '\n';
+	}
+
+} // namespace Debug
+#endif
+
+namespace Require
+{
+	// The next character after the string content should always be the null character.
+	template <typename CharT, std::size_t kN>
+	static void StringNullTerminated(Common::String<CharT, kN> const& buffer)
+	{
+		INFO('\"' << buffer.GetData().data() << "\" with string length " << buffer.GetLength()
+					 << " is not null terminated correctly. "
+					 << "The character at index " << buffer.GetLength() << " is \'"
+					 << buffer.GetData().at(buffer.GetLength()) << "\'.");
+		REQUIRE(buffer.GetData().at(buffer.GetLength()) == '\0');
+	}
+
+	template <typename CharT, std::size_t kN>
+	static void ReplaceString(Common::String<CharT, kN>& buffer,
+									  typename Common::String<CharT, kN>::Data::size_type const index,
+									  std::string_view const string)
+	{
+		INFO("Attempt to replace position " << index << " with \"" << string << "\".");
+		auto const originalStringLength{ buffer.GetLength() };
+		for (std::string_view::size_type count{ 0u }; count < string.length(); ++count)
+		{
+			auto const characterToRemove{ buffer.GetData().at(index) };
+			auto const characterToCopyShiftLeft{ buffer.GetData().at(index + 1u) };
+			INFO("On attempt to remove single character.");
```

Comment:
There are a lot of `INFO` statements here because I was using that information to debug an issue with the window input buffer, and I didn't remove the `INFO` statements afterwards.

The things logged with `INFO` are only printed to the output if the associated test case fails, even if writing test results to a file with `--out`. Also, here, they are not required for the test cases to function.